src/v3/src/components/InputText/InputText.tsx

```diff
@@ -31,130 +25,58 @@ import {
-  const hasErrors = typeof errors !== 'undefined';
-  // TODO: OKTA-569647 - refactor logic
-  const hintId = hint && `${name}-hint`;
-  const explainId = explain && `${name}-explain`;
-  const ariaDescribedByIds = [describedByIds, hintId, explainId].filter(Boolean).join(' ')
-    || undefined;
-  const tokens = useOdysseyDesignTokens();
+  const { errorMessage, errorMessageList } = buildFieldLevelErrorMessages(errors);
 
   return (
-    <Box>
-      <InputLabel
-        htmlFor={name}
-        // To prevent asterisk from shifting far right
-        sx={{ justifyContent: showAsterisk ? 'flex-start' : undefined }}
-      >
-        {label}
-        {showAsterisk && (
-          <Box
-            component="span"
-            sx={{
-              marginInlineStart: tokens.Spacing2,
-              marginInlineEnd: tokens.Spacing2,
-            }}
-            className="no-translate"
-            aria-hidden
-          >
-            *
-          </Box>
-        )}
-        {required === false && (
-          <Typography
-            variant="subtitle1"
-            sx={{ whiteSpace: 'nowrap' }}
-          >
-            {optionalLabel}
-          </Typography>
-        )}
-      </InputLabel>
-      {
-        hint && (
-          <FormHelperText
-            id={hintId}
-            className="o-form-explain"
-            data-se={hintId}
-            // TODO: OKTA-577905 - Temporary fix until we can upgrade to the latest version of Odyssey
-            sx={{ textAlign: 'start' }}
-          >
-            {hint}
-          </FormHelperText>
-        )
-      }
-      <InputBase
-        value={value}
-        type={type || 'text'}
-        id={name}
-        name={name}
-        error={hasErrors}
-        onChange={(e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
-          handleChange?.(e.currentTarget.value);
-        }}
-        onBlur={(e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
-          handleBlur?.(e?.currentTarget?.value);
-        }}
-        disabled={loading}
-        fullWidth
-        inputProps={{
-          'data-se': dataSe,
-          'aria-describedby': ariaDescribedByIds,
-          ...attributes,
-        }}
-        inputRef={focusRef}
-        dir={dir}
-      />
-      {hasErrors && (
-        <FieldLevelMessageContainer
-          messages={errors}
-          fieldName={name}
-        />
-      )}
-      {
-        explain && (
-          <FormHelperText
-            id={explainId}
-            className="o-form-explain"
-            data-se={explainId}
-            // TODO: OKTA-577905 - Temporary fix until we can upgrade to the latest version of Odyssey
-            sx={{ textAlign: 'start' }}
-          >
-            {parsedExplainContent}
-          </FormHelperText>
-        )
-      }
-    </Box>
+    <TextField
+      autoCompleteType={autocomplete}
+      errorMessage={errorMessage}
+      errorMessageList={errorMessageList}
+      hint={hint ?? parsedExplainContent as string}
```

Comment:
`parsedExplainContent` is only used for a single type of input that shows a description for a secondary email address during profile enrollment. `hint` is only used on the primary identify screen if users pass in a custom string for `primaryauth.password.placeholder`. As a result, these two strings are mutually exclusive and can both use the Odyssey `hint` prop

Reply:
It is desired for explanation text to be rendered above the text input field (instead of below the field as was before migration), right?

<img width="460" alt="Screenshot 2024-02-02 at 12 23 45" src="https://github.com/okta/okta-signin-widget/assets/72614880/0dbe1ca0-4be1-4b77-a057-10af66acc093">

Reply:
Casting `parsedExplainContent` to string seems a workaround. In the screenshot above `hint` value is an array of React elements, but Ody expcts string

Reply:
For the explanation text above the field, yes we decided to move it to reuse the `hint` prop location because there's no support for a bottom explain location and we only need it for this one case. For the casting of `parsedExplainContent`, yes I think we are forced to cast as a workaround because Odyssey doesn't allow arrays of React elements. I'm not sure we have another option besides asking Odyssey to support this unless you have other ideas?